manual/library/builtin.etex

```diff
@@ -190,7 +192,10 @@ exception Failure of string
 \index{Failure@\verb`Failure`}
 \begin{ocamldocdescription}
    Exception raised by library functions to signal that they are
-   undefined on the given arguments.
+   undefined on the given arguments.  The string gives some information
+   to the programmer; it is \emph{bad practice} to pattern match
```

Comment:
If the core team agree, I do not mind to be as strong as “_does not_”.

About the fact that it should be frowned upon, we can hardly say that while the standard library makes extensive use of it.  Moreover, `result` is impractical without the use of appropriate monadic connectors — which again are not provided by the standard library.